src/inc/til/env.h

```diff
@@ -0,0 +1,515 @@
+            HRESULT GetShortPathNameW(PCWSTR file, string_type& path)
+            {
+                const auto hr = wil::AdaptFixedSizeToAllocatedResult<string_type, stackBufferLength>(path,
+                                                                                                     [&](_Out_writes_(valueLength) PWSTR value, size_t valueLength, _Out_ size_t* valueLengthNeededWithNull) -> HRESULT {
+                                                                                                         // Note that GetShortPathNameW() is not limited to MAX_PATH
+                                                                                                         // but it does take a fixed size buffer.
+                                                                                                         *valueLengthNeededWithNull = ::GetShortPathNameW(file, value, static_cast<DWORD>(valueLength));
+                                                                                                         RETURN_LAST_ERROR_IF(*valueLengthNeededWithNull == 0);
+                                                                                                         WI_ASSERT((*value != L'\0') == (*valueLengthNeededWithNull < valueLength));
+                                                                                                         if (*valueLengthNeededWithNull < valueLength)
+                                                                                                         {
+                                                                                                             (*valueLengthNeededWithNull)++; // it fit, account for the null
+                                                                                                         }
+                                                                                                         return S_OK;
+                                                                                                     });
+                return hr;
+            }
+
+#ifdef WIL_ENABLE_EXCEPTIONS
+            //! A strongly typed version of the Win32 API GetShortPathNameW.
+            //! Return a path in an allocated buffer for handling long paths.
+            template<typename string_type = wil::unique_cotaskmem_string, size_t stackBufferLength = 256>
+            string_type GetShortPathNameW(PCWSTR file)
+            {
+                string_type result;
+                THROW_IF_FAILED((GetShortPathNameW<string_type, stackBufferLength>(file, result)));
+                return result;
+            }
+#endif
+        };
+    };
+
+    class env
+    {
+    private:
+#ifdef UNIT_TESTING
+        friend class EnvTests;
+#endif
+
+        std::map<std::wstring, std::wstring, til::details::wstring_case_insensitive_compare> _envMap{};
+
+        // these wstring_views better be null terminated.

```

Comment:
The comment is out of date, but we should note that we're taking copies of strings to be certain that they are null-terminated.